python/lsst/pipe/base/pipelineIR.py

```diff
@@ -791,18 +793,52 @@ def from_string(cls, pipeline_string: str):
         return cls(loaded_yaml)
 
     @classmethod
-    def from_file(cls, filename: str):
+    def from_file(cls, filename: str) -> PipelineIR:
         """Create a `PipelineIR` object from the document specified by the
         input path.
 
         Parameters
         ----------
         filename : `str`
             Location of document to use in creating a `PipelineIR` object.
+
+        Returns
+        -------
+        pipelineIR : `PipelineIR`
+            The loaded pipeline
+
+        Note
+        ----
+        This method is deprecated, please use from_uri
         """
-        with open(filename, 'r') as f:
-            loaded_yaml = yaml.load(f, Loader=PipelineYamlLoader)
-        return cls(loaded_yaml)
+        return cls.from_uri(filename)
+
+    @classmethod
+    def from_uri(cls, uri: str) -> PipelineIR:
+        """Create a `PipelineIR` object from the document specified by the
+        input uri.
+
+        Parameters
+        ----------
+        uri: `str`
+            Location of document to use in creating a `PipelineIR` object.
+        Returns
+        -------
+        pipelineIR : `PipelineIR`
+            The loaded pipeline
+        """
+        loadedUri = ButlerURI(uri)
```

Comment:
Really nitpicky, but I'll just note that you're using snake_case for method names here, camelCase for similar method names in Pipeline, camelCase with Biglittle acronyms for "Uri" (while @timj used BigBig for "URI"), and camelCase with BigBig acronyms for "IR".  You could at least use snake_case for variable names, too, now, and avoid at least _some_ of that inconsistency without any API changes.

Ok, so maybe it's not _that_ nitpicky, but also not something we can solve just on this ticket: I am in particular worried about us adding `fromUri`, `fromURI`, or `from_uri` (or similar) to ton of new things over the next few months with absolutely no consistency as we ButlerURI All The Things, and that actually is a real problem for users.

And if we are going to rebel against the dev guide's [_existing_ mandate](https://developer.lsst.io/cpp/style.html?highlight=acronym#id39) that acronyms should Biglittle (`Psf`, `Wcs`, even `Lsst`) - a rebellion I would support, but don't want to lead - we really should RFC it, not just ignore it.

Reply:
So this is a bit of a quirk of how things evolved over time. The snake vs camel came from pipelineIR when it was going to be more of a stand alone thing, and I thought we were moving toward that for new stuff. Then later this got more integrated with Pipeline.py, but there the convention was already camelCase, and so it was not ok to mix the two. So I am not quite sure what we want to do here. pipelineIR is basically all internal, so we could change this around w/o a lot of disruption. If we go this way I would prefer to do the other interface changes on a different ticket to do that. I am so conditioned to camelCase that I keep falling back to it on variable names. I will address any new ones here.

As far as URI vs Uri I tried to emulate the methods that Tim added in Quantum with the mixed case (and I think I missed that with one fromURI) So I guess the question remains, what would you prefer to see done? Maybe we could come up with a plan from there.

Reply:
@timj, do you want to make a call on how we capitalize and separate words in methods involving "URI" going forward?

Once we've settled where we're going, I think I'm basically happy for this PR to choose that or leave existing methods the same as @natelust thinks is best.

Reply:
We are at the confluence of competing naming forces. I'm fine if pipe_base switches to `xxx_uri` naming convention. I've always found `xxxUri` to be a bit odd but I realize that's the old convention.

Reply:
For files that are already camelCase, would you prefer xxxUri or xxxURI? Or would you two be happy if I migrated some interfaces to snake case on this ticket, and others at a different time?

Reply:
I'm fine with incremental migration if it doesn't have much impact or take a lot of extra time. `xxxURI` should be used for classes that are related to `ButlerURI` otherwise things get odd but you are going to say they are already odd.

Reply:
This is more about method names, but if you two are fine with starting migration to snake cake I am fine with that.

Reply:
Yeah, if there isn't an obvious best way to stick with an old style, better to jump all the way to snake case as the new style than land on something awkwardly in-between.